container/common/fsHandler.go

```diff
@@ -106,23 +106,26 @@ func (fh *realFsHandler) update() error {
 }
 
 func (fh *realFsHandler) trackUsage() {
-	fh.update()
+	periodAdjuster := func() {
+		if err := fh.update(); err != nil {
+			klog.Errorf("failed to collect filesystem stats - %v", err)
+			fh.period = fh.period * 2
+			if fh.period > maxBackoffFactor*fh.minPeriod {
+				fh.period = maxBackoffFactor * fh.minPeriod
+			}
+		} else {
+			fh.period = fh.minPeriod
+		}
+	}
+	periodAdjuster()
 	longOp := time.Second
 	for {
 		select {
 		case <-fh.stopChan:
 			return
 		case <-time.After(fh.period):
 			start := time.Now()
-			if err := fh.update(); err != nil {
-				klog.Errorf("failed to collect filesystem stats - %v", err)
-				fh.period = fh.period * 2
-				if fh.period > maxBackoffFactor*fh.minPeriod {
-					fh.period = maxBackoffFactor * fh.minPeriod
-				}
-			} else {
-				fh.period = fh.minPeriod
-			}
+			periodAdjuster()
```

Comment:
Can we instead just move this logic to the top of the loop?  I.E.
```golang
for {
    ... all of the logic ...
    select {
    case <-fh.stopChan:
        return
    case <-time.After(fh.period):
    }
}
```

Also, lets include the logging for long operations in there as well.

Reply:
@dashpole 
Done.